Service/Product/MissingPriceIndexHandler.php

```diff
@@ -0,0 +1,193 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Service\Product;
+
+use Algolia\AlgoliaSearch\Helper\Logger;
+use Magento\Catalog\Model\ResourceModel\Product\Collection as ProductCollection;
+use Magento\Catalog\Model\ResourceModel\Product\CollectionFactory;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\DB\Select;
+use Magento\Framework\Indexer\IndexerInterface;
+use Magento\Framework\Indexer\IndexerRegistry;
+use Magento\Framework\Indexer\StateInterface;
+use Zend_Db_Select;
+
+class MissingPriceIndexHandler
+{
+    public const PRICE_INDEX_TABLE = 'catalog_product_index_price';
+    public const PRICE_INDEX_TABLE_ALIAS = 'price_index';
+    public const MAIN_TABLE_ALIAS = 'e';
+
+    protected array $_indexedProducts = [];
+
+    protected IndexerInterface $indexer;
+    public function __construct(
+        protected CollectionFactory $productCollectionFactory,
+        protected ResourceConnection $resourceConnection,
+        protected Logger $logger,
+        IndexerRegistry $indexerRegistry
+    )
+    {
+        $this->indexer = $indexerRegistry->get('catalog_product_price');
+    }
+
+    /**
+     * @param string[]|ProductCollection $products
+     * @return string[] Array of product IDs that were reindexed by this repair operation
+     * @throws \Zend_Db_Select_Exception
+     */
+    public function refreshPriceIndex(array|ProductCollection $products): array
+    {
+        $reindexIds = $this->getProductIdsToReindex($products);
+        if (empty($reindexIds)) {
+            return [];
+        }
+
+        $this->logger->log(__("Pricing records missing or invalid for %1 product(s)", count($reindexIds)));
+        $this->logger->log(__("Reindexing product ID(s): %1", implode(', ', $reindexIds)));
+
+        $this->indexer->reindexList($reindexIds);
+
+        return $reindexIds;
```

Comment:
Is it planned to use those ids at some point ? Doesn't seem to be the case in the `DataHelper`.

Reply:
Possibly - I thought it might be useful for future reporting either via logs or returning to a UI where we report missing product prices - perhaps the manual SKU reindexer. Return values don't always need to be used but they can provide utility later.